server/Arcadia.Assistant.CSP/Vacations/VacationsPersistenceExecutor.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace Arcadia.Assistant.CSP.Vacations
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Microsoft.EntityFrameworkCore;
+
+    using Arcadia.Assistant.CSP.Model;
+
+    public class VacationsPersistenceExecutor
```

Comment:
why don't you have a number of methods instead? AddApproval, CancelVacation, RejectVacation? I don't really like the current workflow, you can probably make it much easier?

Reply:
Refactored in attempts to make it more clear.